weblab_cg/_generator_chaste_cg.py

```diff
@@ -196,6 +197,42 @@ def _add_units(self):
                 stim_units[key][i]['units'] = getattr(self._model.units.ureg, stim_units[key][i]['units'])
         return stim_units
 
+    def _get_time_variable(self):
+        time_variable = self._model.get_free_variable_symbol()
+        desired_units = self._model.units.ureg.millisecond
+        assert self._model.units.summarise_units(time_variable).dimensionality == desired_units.dimensionality, \
+            'Incorrect definition of time variable (units of time needs to be demensionally equivalent to second)'
+        # If the variable is in units that can be converted to millimolar, perform conversion
+        return self._model.convert_variable(time_variable, desired_units, DataDirectionFlow.INPUT)
+
+    def _get_membrane_voltage_var(self):
+        """ Find the membrane_voltage variable"""
+        voltage = self._model.get_symbol_by_ontology_term(self._OXMETA, "membrane_voltage")
+        desired_units = self._model.units.ureg.millivolt
+        assert self._model.units.summarise_units(voltage).dimensionality == desired_units.dimensionality, \
+            'Incorrect definition of membrane_voltage variable '\
+            '(units of membrane_voltage needs to be demensionally equivalent to Volt)'
+        # Convert if necessary
+        return self._model.convert_variable(voltage, desired_units, DataDirectionFlow.INPUT)
+
+    def _get_cytosolic_calcium_concentration_var(self):
+        """ Find the cytosolic_calcium_concentration variable if it exists"""
+        try:
+            cytosolic_calcium_concentration = \
+                self._model.get_symbol_by_ontology_term(self._OXMETA, "cytosolic_calcium_concentration")
+            desired_units = self._model.units.ureg.millimolar
+
+            # Convert if necessary
+            try:
+                return self._model.convert_variable(cytosolic_calcium_concentration,
+                                                    desired_units, DataDirectionFlow.INPUT)
+            except DimensionalityError:
+                # units can't be converted to millimolar (e.g. could be dimensionless)
+                return cytosolic_calcium_concentration
+        except KeyError:
```

Comment:
I'd move this `except` up before you define `desired_units`. Then you don't need to nest another try-except inside the `try` clause, it can come out a level.